### PR TITLE
Fix slides saving in different folder issue and add professor name to lecture folder name

### PR DESCRIPTION
**Pull request by HemanthK-12**
- Fixes slides saving to same folder issue by replacing the hardcoded `./slides` statement with `config.downloadlocation`, which stores the path till the subject folder location ( after the `CreateDirInsideDownloads` function call).
- Added Professor Name to the folder name by adding the value `courses[choice-1].ProfessorName` during name creation.
- Added comments at places where there are calls to functions in different files, to improve code navigation and readability,
- Fixes Issue #28 .

### CODE REVIEW

File: impartus.go [hunk 1]
```diff
@@ -464,11 +464,11 @@ func CreateTempM3U8File(downloadedPlaylist DownloadedPlaylist) M3U8File {
...
-#EXT-X-VERSION:3
-#EXT-X-MEDIA-SEQUENCE:0
-#EXT-X-ALLOW-CACHE:YES
-#EXT-X-TARGETDURATION:11
-#EXT-X-KEY:METHOD=NONE`)
+						       #EXT-X-VERSION:3
+							   #EXT-X-MEDIA-SEQUENCE:0
+							   #EXT-X-ALLOW-CACHE:YES
+							   #EXT-X-TARGETDURATION:11
+							   #EXT-X-KEY:METHOD=NONE`)
```
Comment (pnicto) [inline impartus.go:471]: Undo this change. As this is a raw string, the output file will also contain the tabs that are part of this string.

Output file after this change:
image

Previous file output:
image

This technically does not cause any bugs, but I prefer the old output.

File: impartus.go [hunk 2]
```diff
@@ -488,11 +488,11 @@ func CreateTempM3U8File(downloadedPlaylist DownloadedPlaylist) M3U8File {
...
-#EXT-X-VERSION:3
-#EXT-X-MEDIA-SEQUENCE:0
-#EXT-X-ALLOW-CACHE:YES
-#EXT-X-TARGETDURATION:11
-#EXT-X-KEY:METHOD=NONE`)
+								#EXT-X-VERSION:3
+								#EXT-X-MEDIA-SEQUENCE:0
+								#EXT-X-ALLOW-CACHE:YES
+								#EXT-X-TARGETDURATION:11
+								#EXT-X-KEY:METHOD=NONE`)
```
Comment (pnicto) [inline impartus.go:495]: Undo this change. As this is a raw string, the output file will also contain the tabs that are part of this string.

Output file after this change:
image

Previous file output:
image

This technically does not cause any bugs, but I prefer the old output.

**Review by pnicto — CHANGES_REQUESTED** [review]
Can you undo all the comments? I don't think you wanted to push the comments(those that indicate the filename eg., `in parser.go` etc.,)

File: impartus.go (hunk 1 above)
Comment (HemanthK-12) [reply impartus.go:471]: yeah sure, will do that.

File: impartus.go (hunk 2 above)
Comment (HemanthK-12) [reply impartus.go:495]: yeah sure, will do that.

File: impartus.go [hunk 3]
```diff
@@ -192,7 +192,7 @@ func GetCourses() Courses {
 	log.Println("Getting courses")
 
 	var courses Courses
-	config := GetConfig()
+	config := GetConfig()
```
Comment (pnicto) [inline impartus.go:195]:
```suggestion
	config := GetConfig()
```

**Review by pnicto — CHANGES_REQUESTED** [review]
All good

**Review by pnicto — APPROVED** [review]
Good work! And thanks for your contribution 😄
